src/Streamly/Internal/Data/Stream/IsStream/Transform.hs

```diff
@@ -1666,3 +1668,22 @@ applyAsync = (|$)
 x |& f = f |$ x
 
 infixl 1 |&
+
+-- | Transform a stream of @a@ to @(Just a)@
+-- /Internal/
+--
+{-# INLINE justsOf #-}
+justsOf :: (MonadIO m, IsStream t)
+    => t m a -> t m (Maybe a)
+justsOf = map Just
+
+-- | Transform a stream of @a@ to @(Just a)@ then intersperse a Nothing into the input
+-- stream after every n elements then interject Nothing every @t@ seconds.
+-- @since 0.8.0
+{-# INLINE justsOfTimeout #-}
+justsOfTimeout :: (MonadIO m, IsStream t, MonadAsync m)
+    => Int -> Double -> t m a -> t m (Maybe a)
+justsOfTimeout n t =
+      interjectSuffix t (return Nothing)
+    . intersperseSuffixBySpan n (return Nothing)
+    . justsOf
```

Comment:
As mentioned in #456 this has an undesired behavior. The timeout may occur after the span is emitted and the timeout would force eject a smaller span. Ideally we should reset the timer whenever we emit a span.